src/main/java/net/dries007/tfc/objects/blocks/plants/BlockPlantTFC.java

```diff
@@ -201,22 +286,47 @@ public boolean canBlockStay(World worldIn, BlockPos pos, IBlockState state)
     @Nonnull
     public AxisAlignedBB getBoundingBox(IBlockState state, IBlockAccess source, BlockPos pos)
     {
-        return super.getBoundingBox(state, source, pos).offset(state.getOffset(source, pos));
+        return PLANT_AABB.offset(state.getOffset(source, pos));
     }
 
     @Override
     @Nonnull
     public net.minecraftforge.common.EnumPlantType getPlantType(net.minecraft.world.IBlockAccess world, BlockPos pos)
     {
-        IBlockState iblockstate = world.getBlockState(pos.down());
-        if (plant.getPlantType() == Plant.PlantType.DESERT && BlocksTFC.isSand(iblockstate))
-        {
-            return EnumPlantType.Desert;
-        }
-        else
-        {
-            return EnumPlantType.Plains;
-        }
+        if (plant.getPlantType() == Plant.PlantType.DESERT) return EnumPlantType.Desert;
+        if (plant.getPlantType() == Plant.PlantType.DESERT_TALL_PLANT) return EnumPlantType.Desert;
+        if (plant.getPlantType() == Plant.PlantType.CACTUS) return EnumPlantType.Desert;
+        if (plant.getPlantType() == Plant.PlantType.FLOATING) return EnumPlantType.Water;
+        if (plant.getPlantType() == Plant.PlantType.FLOATING_SEA) return EnumPlantType.Water;
+        if (plant.getPlantType() == Plant.PlantType.MUSHROOM) return EnumPlantType.Cave;
+        return EnumPlantType.Plains;
+    }
+
+    @Nonnull
+    public Plant.EnumPlantTypeTFC getPlantTypeTFC()
+    {
+        if (plant.getIsClayMarking()) return Plant.EnumPlantTypeTFC.Clay;
```

Comment:
This could either be a switch statement, or build it into `Plant.PlantType` (i.e. have each `PlantType` hold a private `EnumPlantTypeTFC`, accessed with a `Plant.PlantType#getEnumType())` or similar